binary_gentoo/internal/cli/tree_diff.py

```diff
@@ -3,12 +3,47 @@
 
 import filecmp
 import os
+import re
 import sys
 from argparse import ArgumentParser
+from typing import Set
 
-from ..reporter import exception_reporting
+from ..reporter import announce_and_check_output, exception_reporting
 from ._parser import add_version_argument_to
 
+_keywords_pattern = re.compile('KEYWORDS="(?P<keywords>[^"]*)"')
+_filename_9999_pattern = re.compile(r'9999(\-r[0-9]+)?\.ebuild$')
+
+
+def _replace_special_keywords_for_ebuild(accept_keywords: Set[str],
+                                         ebuild_keywords: Set[str]) -> Set[str]:
+    if '**' in accept_keywords:
+        accept_keywords.remove('**')
+        accept_keywords |= ebuild_keywords
+    elif '*' in accept_keywords:
+        accept_keywords.remove('*')
+        accept_keywords |= {kw for kw in ebuild_keywords if not kw.startswith('~')}
+    elif '~*' in accept_keywords:
+        accept_keywords.remove('~*')
+        accept_keywords |= {kw for kw in ebuild_keywords if kw.startswith('~')}
+    return accept_keywords
+
+
+def _get_relevant_keywords_set_for(ebuild_file: str, accept_keywords: Set[str]) -> Set[str]:
+    with open(ebuild_file) as ifile:
```

Comment:
With my picky hat on:
This line made me notice that "file" is used for both file "handles" and filenames. Would you mind renaming the filename ones to say "_filename" rather than just "_file"?

Reply:
I renamed a few variables to "_filepath", if that is OK with you too.

Reply:
Sounds good, thanks!